### ISSUE TITLE
Extending a plugin

### ISSUE THREAD

**Issue by lww** [kasperisager/vanilla-api#78]
Hi, is it possible to extend the API from a plugin?

I have this Upload Editor plugin `class EditorPlugin extends Gdn_Plugin` and would like to point to this function: 
```
@param PostController $Sender
@param array $Args

public function postController_editorUpload_create($Sender, $Args = array()) {}
```
Basically, what I want to achieve is to get "/post/editorupload" to work with your API...

**Comment by kasperisager** [comment]
You can find documentation on extending the API here: https://github.com/kasperisager/vanilla-api/wiki/Extending. What I'm guessing you'd need to specify are the `controller` and `method` arguments:

```php
...
"controller" => "Post",
"method" => "editorUpload"
...
```

You should definitely leave out `application` and the rest of the arguments are optional.

**Comment by lww** [comment]
Thanks for your reply.

To test things out I created a sample plugin:
```
class Foo extends Gdn_Plugin {
   public function discussionsController_kaboom_create($Sender) {
        echo "Kaboom!";
   }
}
```

 A html GET request to /discussions/kaboom delivers correctly "Kaboom!".

Then I added this to your API's **class.discussionsapi.php**:
```
final class DiscussionsAPI extends APIMapper
{
  ...
  public static function register($data)
  {
    ...
    static::get("/kaboom", [
      "controller" => "Discussions",
      "method" => "kaboom"
    ]);
    ...
  }
} 
```

JSON request to **/api/discussions/kaboom** raises an error:
```
{
  "Code": 405,
  "Exception": "API.Error.MethodNotAllowed"
}
```

Are you sure the API is handling plugins correctly?

**Comment by kasperisager** [comment]
A status of 405 could indicate that the mapped method hasn't been registered. Have you tried clearing Vanilla's cache?

**Comment by lww** [comment]
Yes, I already cleared the cache. No difference, still status 405...

It has to be an issue with plugins, because it works if a add a method from an existing controller to the api.

**Comment by kasperisager** [comment]
Gotcha. The application uses the built-in `Gdn::request()->withControllerMethod()` for dispatching requests so I'm guessing that doesn't work with virtual controller methods 😕

**Comment by lww** [comment]
That's what I suspected. Thanks for the clarification! :thumbsup:

**Comment by masiorama** [comment]
@lww Hey there, sorry to bother, but I'm trying to achieve something similar, I read the guide on how to extend but I'm a little lost on how to mix everything up.
My task is very simple, I want to get the last n registered user.
I tried adding 
`static::get("/last", [
            "controller" => "User",
            "method" => "last"
        ]);`
to class.usersapi.php and then created a plugin with the method:
`public function usersController_last_create($Sender)`

I don't know what to do now. Sorry, I just approached vanilla so I'm a bit lost. Any suggestion would be very appreciated. Thanks!
